controller/src/main/java/org/jboss/as/controller/operations/global/MapOperations.java

```diff
@@ -155,6 +157,9 @@ private MapRemoveHandler() {
         void updateModel(final OperationContext context, ModelNode model, AttributeDefinition attributeDefinition, ModelNode attribute) throws OperationFailedException {
             String key = KEY.resolveModelAttribute(context, model).asString();
             attribute.remove(key);
+            if(attribute.asPropertyList().isEmpty()){
```

Comment:
bit faster way would be attribute.asList() also would avoid some possible issues with type of model